### PR TITLE
Add Rendering Plugin Tutorial

### PR DESCRIPTION
**Pull request by JShep1**
Adding a more high-leveled tutorial of how to create a custom rendering engine plugin.  Should we go into more detail as the [physics equivalent](https://ignitionrobotics.org/api/physics/3.0/physicsplugin.html) does?  

Signed-off-by: John Shepherd <john@openrobotics.org>

### CODE REVIEW

File: tutorials/03_rendering_plugins.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+\page renderingplugin Understanding the Rendering Plugin
+
+This is an introduction to different rendering engines and how they are integrated into the Ignition Physics library.
```
Comment (iche033) [inline tutorials/03_rendering_plugins.md:3]: `Ignition Physics` -> `Ignition Rendering`

File: tutorials/03_rendering_plugins.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * Registering the plugin, this line can be seen at the bottom of the [`HelloWorldPlugin`](https://github.com/ignitionrobotics/ign-rendering/tree/ign-rendering3/examples/hello_world_plugin/HelloWorldPlugin.cc) example
+
+Finally, for your custom rendering engine to actually have any functionality, you will need to implement your own `Scene` class, which inherits from and implements the pure virtual functions of [`ignition::rendering::Scene`](https://github.com/ignitionrobotics/ign-rendering/blob/ign-rendering3/include/ignition/rendering/Scene.hh).  The mocked example simply returns `nullptr` for its `CreateSceneImpl(...)` function, so it may be useful to look at the current `Scene` implementations for the other rendering engines within `ignition::rendering` such as [`ogre`](https://github.com/ignitionrobotics/ign-rendering/blob/ign-rendering3/ogre/src/OgreScene.cc) or [`optix`](https://github.com/ignitionrobotics/ign-rendering/blob/ign-rendering3/optix/src/OptixScene.cc).
+
+#### Building and Running Your Rendering Plugin
```
Comment (iche033) [inline tutorials/03_rendering_plugins.md:23]: can you append ` with Ignition Gazebo` to this heading? 

I think there are potentially others who are interested in building their own render engine plugin without requiring ignition gazebo to run it. That's doable and probably deserves its own example and tutorial.

File: tutorials/03_rendering_plugins.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ * Registering the plugin, this line can be seen at the bottom of the [`HelloWorldPlugin`](https://github.com/ignitionrobotics/ign-rendering/tree/ign-rendering3/examples/hello_world_plugin/HelloWorldPlugin.cc) example
+
+Finally, for your custom rendering engine to actually have any functionality, you will need to implement your own `Scene` class, which inherits from and implements the pure virtual functions of [`ignition::rendering::Scene`](https://github.com/ignitionrobotics/ign-rendering/blob/ign-rendering3/include/ignition/rendering/Scene.hh).  The mocked example simply returns `nullptr` for its `CreateSceneImpl(...)` function, so it may be useful to look at the current `Scene` implementations for the other rendering engines within `ignition::rendering` such as [`ogre`](https://github.com/ignitionrobotics/ign-rendering/blob/ign-rendering3/ogre/src/OgreScene.cc) or [`optix`](https://github.com/ignitionrobotics/ign-rendering/blob/ign-rendering3/optix/src/OptixScene.cc).
```
Comment (iche033) [inline tutorials/03_rendering_plugins.md:21]: I would remove the optix link as it not really being maintained right now.

Comment (iche033) [inline tutorials/03_rendering_plugins.md:21]: Maybe worth mentioning that, at minimum, to have something displayed in a window, developers would need to implement a Scene class and also a Camera class. The Camera class does the actual rendering to generate the images (of the scene) to be displayed.

File: tutorials/03_rendering_plugins.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+\note You can use the `--render-engine`, `--render-engine-server`, and `--render-engine-gui` flags to specify any supported or custom rendering engine at runtime, this flag will override any other pre-existing rendering engine specifications (such as in an `.sdf` file).  The command line call would look something like:
+
+~~~
+ign gazebo --render-engine HelloWorldPlugin shapes.sdf
+~~~
+
+#### Feature Comparison
+
+Do we want to fill this out for ogre, ogre2, and optix?
```
Comment (iche033) [inline tutorials/03_rendering_plugins.md:39]: I've been wanting to do this for some time. We can either add to this PR or in a separate PR. I'll let you decide but yes this is definitely useful.

File: tutorials/03_rendering_plugins.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+This is an introduction to different rendering engines and how they are integrated into the Ignition Physics library.
+
+## Ignition Rendering
+
+The \ref ignition::rendering "Ignition Rendering" library integrates external rendering engines into the Ignition Simulation eco-system.
+It allows users to select from multiple supported rendering engines based on their simulation needs.
+Its plugin interface loads rendering engines at runtime.
+It is also possible to integrate your own selected rendering engine by writing a compatible plugin interface.
+
+#### How to Write Your Own Rendering Plugin
```
Comment (iche033) [inline tutorials/03_rendering_plugins.md:12]: `...Rendering Engine Plugin`

File: tutorials/03_rendering_plugins.md (hunk 1 above)
Comment (JShep1) [reply tutorials/03_rendering_plugins.md]: 86be9fc553ccaa6a7994609c581be9d3e3b7e649

File: tutorials/03_rendering_plugins.md (hunk 2 above)
Comment (JShep1) [reply tutorials/03_rendering_plugins.md:23]: 86be9fc553ccaa6a7994609c581be9d3e3b7e649

File: tutorials/03_rendering_plugins.md (hunk 3 above)
Comment (JShep1) [reply tutorials/03_rendering_plugins.md]: 86be9fc553ccaa6a7994609c581be9d3e3b7e649

Comment (JShep1) [reply tutorials/03_rendering_plugins.md]: 86be9fc553ccaa6a7994609c581be9d3e3b7e649

File: tutorials/03_rendering_plugins.md (hunk 4 above)
Comment (JShep1) [reply tutorials/03_rendering_plugins.md]: Sounds good, let's add it in a separate PR

File: tutorials/03_rendering_plugins.md (hunk 5 above)
Comment (JShep1) [reply tutorials/03_rendering_plugins.md]: 86be9fc553ccaa6a7994609c581be9d3e3b7e649

**Review by iche033 — APPROVED** [review]
made one minor tweak to heading. 128638f6ff657889a0b029cf3c03c25c0b256278

This looks good to me.

**Pull request merged**